core/src/focus_tracker.rs

```diff
@@ -110,6 +148,43 @@ impl<'gc> FocusTracker<'gc> {
 
         if next.is_some() {
             self.set(next.copied(), context);
+            self.update_highlight();
+        }
+    }
+
+    fn update_highlight(&self) {
+        self.0.highlight.replace(self.redraw_highlight());
+    }
+
+    fn redraw_highlight(&self) -> Highlight {
+        let Some(focus) = self.get() else {
+            return Highlight::Inactive;
+        };
+
+        if !focus.is_highlight_enabled() {
+            return Highlight::Inactive;
         }
+
+        let bounds = focus.world_bounds().grow(-Self::HIGHLIGHT_WIDTH / 2);
+        let mut drawing = Drawing::new();
```

Comment:
Using Drawing for this is going to be a little expensive on the rendering side, but realistically this isn't going to be hit enough to be noticed (I hope?)
Ideally we have a draw command that's "draw box outline" later, and let the render backend optimise that. It can make it pixel perfect too, which would be nice.

Not something you need to change for this PR - just a comment!